zerver/lib/import_realm.py

```diff
@@ -1303,3 +1306,78 @@ def import_analytics_data(realm: Realm, import_dir: Path) -> None:
     re_map_foreign_keys(data, 'analytics_streamcount', 'stream', related_table="stream")
     update_model_ids(StreamCount, data, 'analytics_streamcount')
     bulk_import_model(data, StreamCount)
+
+    fix_datetime_fields(data, 'analytics_fillstate')
+    handle_fill_state(data, 'analytics_fillstate', realm)
+
+# To Handle FillState we have to compare the state of new (current) server to the old
+# (imported realm) server. If the new server is currently empty, we just import
+# FillState from the old server, if the old server is ahead, we delete any entries
+# from the imported realm's analytics tables and if the new server is ahead, then
+# we have to count stats just for the target Realm.
+def handle_fill_state(data: TableData, table: TableName, realm: Realm) -> None:
+    fill_state = FillState.objects.all()
+    if fill_state is None:
```

Comment:
This is most clearly written `if fill_state.exists()`.

Reply:
It may also be worth testing whether the Analytics code will avoid creating FillState` entries when run with an empty Zulip server (well, empty except for the system bots).  My guess is it won't, so this empty condition will only happen if one imports data into the server before the first `update_analytics_counts` cron run.  Maybe OK to leave this as a follow-up issue since it might be hard to fix without a lot of work, but we should at least document it with a comment.

Reply:
@timabbott Exactly so we have to define a condition where we can say that the server is truly empty.

Reply:
OK.  I think that we should plan to write a function for that check that can live in the new library we'll be adding in https://github.com/zulip/zulip/issues/13736, and then this logic can use that.

I think it can be roughly checking that there's no realm with a name other than `zulipinternal` on the server; feel free to just add a function called `zulip_server_is_initialized_and_empty` at the time of `import_realm.py` doing that and then we can move it once #13736 is merged.

Reply:
OK, we've finished the work required to unblock #13736.  If you're excited about unblocking this, you could claim that issue (I think the technical plan is pretty clear if you want to do it before Mateusz has a chance); it'd be a good opportunity to practice writing a super clear 10-commit series refactor.

Reply:
@timabbott Sure I will take a look at it.